pkg/oauth/oauth.go

```diff
@@ -136,6 +136,13 @@ func (s *server) Authorize(authorizePage echo.HandlerFunc) echo.HandlerFunc {
 			switch c.Request().Method {
 			case http.MethodPost:
 				ar.Authorized, _ = strconv.ParseBool(req.PostForm.Get("authorize"))
+				if !ar.Authorized {
+					events.Publish(evtAccessTokenDeleted.NewWithIdentifiersAndData(req.Context(), &session.UserIdentifiers, nil))
+					err = s.store.DeleteSession(req.Context(), &session.UserIdentifiers, session.SessionId)
+					if err != nil && !errors.IsNotFound(err) {
+						return err
+					}
+				}
```

Comment:
This doesn't look quite right. If on the authorization screen I click 'reject', there shouldn't be an access token to delete, and I shouldn't be logged out either.

Reply:
Indeed, this is the wrong event published there... However, in order to redirect back to the login page, we have to delete the session.